main_test.go

```diff
@@ -339,6 +374,39 @@ func TestServe(t *testing.T) {
 			},
 			startHTTP,
 		},
+		{
+			"http cache max payload size",
```

Comment:
this is a good test but you should also add the test that will asses if when the maxPayloadSize is activated if a payload a  smaller than the limit will be cached